app/models/address.rb

```diff
@@ -30,18 +31,46 @@ def full_address
 
   alias to_s full_address
 
+  def geocode_with_ward
+    # TODO? Is there any point using the Geocoder gem rather than writing
+    # our own code to query postcodes.io given that using postcodes.io
+    # now defines how we use Geocoder?
+    # - Passing anything other than a postcode will cause Geocoder.search to
+    #   return an empty array.
+    # - We are accessing admin_ward directly through through
+    #   Geocoder::Result::PostcodesIo#data hash.
+
+    geo = Geocoder.search(postcode).first&.data
+
+    if geo
+      # TODO? This works. Is there a more correct (railsy) way of doing this
+      # e.g. using write_attribute ? What is the trade off?
+      self.longitude= geo['longitude']
+      self.latitude= geo['latitude']
+      self.admin_ward= geo['admin_ward']
+    end
+  end
+
   class << self
     # location - The raw location field
-    # components - Array containing parts of an event's location field
+    # components - Array containing parts of an event's location field, excluding the postcode.
     def search(location, components, postcode)
+
+      # Find the first Address whose first address line contains any one of the
+      # address lines in the components argument.
```

Comment:
I think find_by is the correct method to use here, could be wrong https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by

Reply:
Although I still wonder whether this is the actual operation we want to perform. If the first line of the new address were "University of Manchester" then an existing address which started with "Skiving Dept, University of Manchester" would be found and used instead.

Reply:
Yeah, will need updating at some point. On the phone I suggested that maybe one thing Places can do it list an address as `canonical` - i.e. that any match will normalise to that one if found?

Reply:
We still need to define 'match' though.

Reply:
Lat/lon check I think right @GabAlexandria?

Reply:
That was a separate check (now commented out) in the original code. Also, by using postcodes.io, everything in the same postcode will have the same latlon.